tools/pydfu.py

```diff
@@ -360,9 +363,13 @@ def get_dfu_devices(*args, **kwargs):
     Additional filters (like idProduct and idVendor) can be passed in to
     refine the search.
     """
-
-    return list(usb.core.find(*args, find_all=True,
-                              custom_match=FilterDFU(), **kwargs))
+    l = []
+    devices = usb.core.find(*args, find_all=True,
+                         custom_match=FilterDFU(), **kwargs)
+    # convert to list for compatibility with newer pyusb
+    for d in devices:
+        l.append(d)
+    return l
```

Comment:
I think you're right. That part of the changeset was made by somebody else and I was just pulling it across.

Let me try to get rid of that change in openmv and if that flies, I'll update this PR. It may take a few days (I'm at a conference all this week).